Automatically add `mailto:` to email addresses when linking

luisherranz opened the pull request:
## Description
<!-- Please describe what you have changed or added -->
The link field is automatically populated when the selected text is an email.

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->
I've done manual tests of the editor and added unit tests for the `isEmail` validator.

## Screenshots <!-- if applicable -->
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/3305402/55684191-b3c41400-5948-11e9-8c72-cecf53f85550.gif)


## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [x] My code is tested (only unit tests for `isEmail`).
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://wordpress.org/gutenberg/handbook/designers-developers/ -->

----

Fixes #9774

luisherranz committed:
Automatically add `mailto:` to email addresses when linking

mtias commented on packages/format-library/src/link/index.js:
```diff
@@ -20,6 +20,8 @@ import InlineLinkUI from './inline';
 
 const name = 'core/link';
 
+const isEmail = ( email ) => /[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/.test( email );
```
This function should probably be added to the `@wordpress/url` package.

mtias replied:
The matcher is there already, actually: https://github.com/WordPress/gutenberg/blob/master/packages/url/src/index.js#L7

luisherranz replied:
Sure.

luisherranz replied:
Added to `@wordpress/url` in https://github.com/WordPress/gutenberg/pull/14857/commits/cb8272af748480b7c6db0cff64154c9e16dcf958 and fixed in  https://github.com/WordPress/gutenberg/pull/14857/commits/6620026a0cc79af5d37b776dc634467cf27d613f.

luisherranz committed:
Add tests for isEmail

Valid and invalid examples from https://en.wikipedia.org/wiki/Email_address#Examples

luisherranz committed:
Add source and updated README

luisherranz committed:
Use isEmail from @wordpress/url

youknowriad reviewed (approved):
Nice, thanks.

mtias commented:
Thanks @luisherranz :)
